pkg/cfn/builder/vpc_existing_test.go

```diff
@@ -0,0 +1,215 @@
+package builder_test
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	awsec2 "github.com/aws/aws-sdk-go/service/ec2"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/cfn/builder"
+	"github.com/weaveworks/eksctl/pkg/cfn/builder/fakes"
+	"github.com/weaveworks/eksctl/pkg/eks/mocks"
+	gfnt "github.com/weaveworks/goformation/v4/cloudformation/types"
+)
+
+var _ = Describe("Existing VPC", func() {
+	var (
+		vpcRs   *builder.ExistingVPCResourceSet
+		cfg     *api.ClusterConfig
+		mockEC2 *mocks.EC2API
+	)
+
+	BeforeEach(func() {
+		cfg = api.NewClusterConfig()
+		cfg.VPC = vpcConfig()
+		cfg.AvailabilityZones = []string{azA, azB}
+		cfg.VPC.ID = "custom-vpc"
+		mockEC2 = &mocks.EC2API{}
+	})
+
+	JustBeforeEach(func() {
+		mockEC2.On("DescribeVpcs", &awsec2.DescribeVpcsInput{
+			VpcIds: aws.StringSlice([]string{"custom-vpc"}),
+		}).Return(&awsec2.DescribeVpcsOutput{
+			Vpcs: []*awsec2.Vpc{
+				{
+					VpcId: aws.String("custom-vpc"),
+					Ipv6CidrBlockAssociationSet: []*awsec2.VpcIpv6CidrBlockAssociation{
+						{
+							Ipv6CidrBlock: aws.String("foo"),
+						},
+					},
+				},
+			},
+		}, nil)
+		vpcRs = builder.NewExistingVPCResourceSet(builder.NewRS(), cfg, mockEC2)
+	})
+
+	Describe("CreateTemplate", func() {
+		var (
+			addErr        error
+			vpcID         *gfnt.Value
+			subnetDetails *builder.SubnetDetails
+			vpcTemplate   *fakes.FakeTemplate
+		)
+
+		JustBeforeEach(func() {
+			vpcID, subnetDetails, addErr = vpcRs.CreateTemplate()
+			vpcTemplate = &fakes.FakeTemplate{}
+			templateBody, err := vpcRs.RenderJSON()
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(json.Unmarshal(templateBody, vpcTemplate)).To(Succeed())
+		})
+
+		It("uses the existing VPC", func() {
+			Expect(addErr).NotTo(HaveOccurred())
+			By("the correct VPC resource values are loaded into the VPCResource")
+			Expect(vpcID).To(Equal(gfnt.NewString("custom-vpc")))
+
+			By("no resources are added to the set")
+			Expect(vpcTemplate.Resources).To(HaveLen(0))
+
+			By("the private subnet resource values are loaded into the VPCResource")
+			Expect(subnetDetails.Private).To(HaveLen(2))
+			Expect(subnetDetails.Private).To(ContainElement(builder.SubnetResource{
+				Subnet:           gfnt.NewString(privateSubnet2),
+				AvailabilityZone: azB,
+			}))
+			Expect(subnetDetails.Private).To(ContainElement(builder.SubnetResource{
+				Subnet:           gfnt.NewString(privateSubnet1),
+				AvailabilityZone: azA,
+			}))
```

Comment:
You can also use `ContainElements` here:
```suggestion
			Expect(subnetDetails.Private).To(ContainElements(
				builder.SubnetResource{
					Subnet:           gfnt.NewString(privateSubnet2),
					AvailabilityZone: azB,
				},
				builder.SubnetResource{
					Subnet:           gfnt.NewString(privateSubnet1),
					AvailabilityZone: azA,
				}),
			)		

```